### PR TITLE
✨ Add grace period to reach LAPI without blocking further queries

### PR DESCRIPTION
**Pull request by mathieuHa**

### CODE REVIEW

File: bouncer.go [hunk 1]
```diff
@@ -279,8 +283,12 @@ func (bouncer *Bouncer) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	// Right here if we cannot join the stream we forbid the request to go on.
 	if bouncer.crowdsecMode == configuration.StreamMode || bouncer.crowdsecMode == configuration.AloneMode {
+		//nolint:gocritic
 		if isCrowdsecStreamHealthy {
 			handleNextServeHTTP(bouncer, remoteIP, rw, req)
+		} else if bouncer.nbFailedStreamUpdate < bouncer.maxFailedStreamUpdate || bouncer.maxFailedStreamUpdate == -1 {
+			bouncer.nbFailedStreamUpdate++
```
Comment (maxlerebourg) [inline bouncer.go:290]: If the `maxFailedStreamUpdate` is set to 2 and 3 people try to access a route, you block the third and all the next ones

File: bouncer.go [hunk 2]
```diff
@@ -517,6 +525,7 @@ func handleStreamCache(bouncer *Bouncer) error {
 	}
 	bouncer.log.Debug("handleStreamCache:updated")
 	isCrowdsecStreamHealthy = true
+	bouncer.nbFailedStreamUpdate = 0
```
Comment (maxlerebourg) [inline bouncer.go:528]: do with a global var like `isCrowdsecStreamHealthy` or replace `isCrowdsecStreamHealthy` by `bouncer.isCrowdsecStreamHealthy`

File: bouncer.go (hunk 1 above)
Comment (mathieuHa) [reply bouncer.go:290]: That's true, we need to increment only once every stream update

File: bouncer.go (hunk 2 above)
Comment (mathieuHa) [reply bouncer.go:528]: if there was no reason to use global var in the first place, i suggest add an attribute for the bouncer

File: bouncer.go [hunk 3]
```diff
@@ -360,14 +360,13 @@ func handleNextServeHTTP(bouncer *Bouncer, remoteIP string, rw http.ResponseWrit
...
 
 func handleStreamTicker(bouncer *Bouncer) {
-	bouncer.log.Debug(fmt.Sprintf("handleStreamTicker:start isCrowdsecStreamHealthy:%t", isCrowdsecStreamHealthy))
 	if err := handleStreamCache(bouncer); err != nil {
-		bouncer.log.Debug(fmt.Sprintf("handleStreamTicker:%d failure to reach crowdsec", updateFailure))
-		if updateFailure >= bouncer.updateMaxFailure && bouncer.updateMaxFailure != -1 {
+		bouncer.log.Debug(fmt.Sprintf("handleStreamTicker updateFailure:%d isCrowdsecStreamHealthy:%t %s", updateFailure, isCrowdsecStreamHealthy, err.Error()))
+		if bouncer.updateMaxFailure != -1 && updateFailure >= bouncer.updateMaxFailure && isCrowdsecStreamHealthy {
 			isCrowdsecStreamHealthy = false
+			bouncer.log.Error(fmt.Sprintf("handleStreamTicker:error updateFailure:%d %s", updateFailure, err.Error()))
```
Comment (mathieuHa) [inline bouncer.go:367]: why not use the same format of logs as just below ?

**Review by maxlerebourg — APPROVED** [review]

**Pull request merged**